### PR TITLE
Feature hydrogen aluminum integration more updates

### PR DESCRIPTION
**Pull request by benson31**
These are the updates to match [#45](https://github.com/LLNL/Elemental/pull/45)

### CODE REVIEW

**Comment by timmoon10** [comment]
Gradient checking on tests/model_mnist_conv_graph.prototext runs as expected.

File: src/callbacks/callback_ltfb.cpp [hunk 1]
```diff
@@ -91,8 +90,27 @@ void exchange_weights(lbann_comm* comm,
...
+        break;
+#ifdef HYDROGEN_HAVE_CUDA
+      case El::Device::GPU:
+        using ValueT
+          = std::remove_pointer<decltype(remote_matrix->Buffer())>::type;
+        comm->sendrecv(
+          local_matrix.LockedBuffer(), size, partner,
+          remote_matrix->Buffer(), size, partner,
+          El::SyncInfo<El::Device::GPU>{
+            static_cast<El::Matrix<ValueT,El::Device::GPU> const&>(
```
Comment (timmoon10) [inline src/callbacks/callback_ltfb.cpp]: This is very general, although maybe overkill. `static_cast<GPUMat const&>` would probably be sufficient.

**Comment by JaeseungYeom** [comment]
I am getting a compilation error:
/usr/workspace/wsb/yeom2/jag-conduit/lbann-tom/include/lbann/comm.hpp:489:52: note:   cannot convert '1' (type 'int') to type 'const int*'
                     El::SyncInfo<El::Device::CPU>{});

**Comment by benson31** [comment]
@JaeseungYeom Is this with gcc 4.9.3? Regardless, this shouldn't be an error. It should match `Gather(const T*, int, T*, int, int, Comm, SyncInfo<D>)`. Can you send me the full error?

**Comment by JaeseungYeom** [comment]
Yes, it was gcc 4.9.3 on surface. I gave you a text file on LC. take it by 'take -a yeom2'

**Comment by benson31** [comment]
Oh shoot. I'll update `build_lbann_lc.sh` for you.

**Comment by benson31** [comment]
Pull and try again.

**Comment by JaeseungYeom** [comment]
That worked. I will start running this now, and let you know how it ran tomorrow.

**Comment by timmoon10** [comment]
I've been running VRAM experiments with https://github.com/LLNL/Aluminum/pull/2, https://github.com/LLNL/Aluminum/pull/3, https://github.com/LLNL/Elemental/pull/49, and this PR. I get similar timings and behavior and Pascal (the new code is ~1% faster). I also get identical results with model_zoo/tests/model_mnist_ridge_regression.prototext (which also performs gradient checking). My vote is to merge the entire lot.

**Pull request merged**